File: level 0/점의 위치 구하기.py
def solution(dot):
    if dot[0] > 0:
        if dot[1] > 0:
            return 1
        else:
            return 4
    if dot[0] < 0:
        if dot[1] > 0:
            return 2
        else:
            return 3

# 다른 사람의 풀이
# def solution(dot):
#     quad = [(3,2),(4,1)]     --> dot[0](x좌표)이 0보다 작은것끼리, 큰것끼리 묶음
#     return quad[dot[0] > 0][dot[1] > 0]     --> []안에 있는 코드가 True면 1이, False면 0이 들어갈 것임.

File: javascript_programmers_school/level 0/점의 위치 구하기.js
function solution(dot) {
  if (dot[0] > 0) {
      const result = dot[1] > 0 ? 1 : 4;
      return result
  }
  else {
      const result = dot[1] > 0 ? 2 : 3;
      return result
  }   
}

// 짧은 풀이 (내 답안을 줄여서 아래와 같이 작성할 수 있다.)
// function solution(dot) {
//   return dot[0] > 0 ? (dot[1] > 0 ? 1 : 4) : (dot[1] > 0 ? 2 : 3);
// }


// 구조분해할당 이용
// function solution(dot) {
//   const [num,num2] = dot;     --> dot의 첫 번째 요소는 num에, 두 번째 요소는 num2에 담기게 됨.
//   const check = num * num2 > 0;     --> 둘 다 양수인지 음수인지 혹은 양수 음수가 섞여있는지 확인하기
//   return num > 0 ? (check ? 1 : 4) : (check ? 3 : 2);
// }